applicationset/controllers/applicationset_controller.go

```diff
@@ -1018,6 +1018,16 @@ func statusStrings(app argov1alpha1.Application) (string, string, string) {
 	return healthStatusString, syncStatusString, operationPhaseString
 }
 
+func getAppStep(appName string, appStepMap map[string]int) string {
```

Comment:
I think for further reusability of this function, it would make more sense to return an `int` instead of a string here.